external-crates/move/tools/move-package/src/resolution/dependency_graph.rs

```diff
@@ -1452,25 +1457,36 @@ fn deps_equal<'a>(
     graph1_pkg_table: &'a BTreeMap<PM::PackageName, Package>,
     graph2: &'a DependencyGraph,
     graph2_pkg_table: &'a BTreeMap<PM::PackageName, Package>,
+    overrides: &'a BTreeMap<PM::PackageName, Package>,
 ) -> std::result::Result<
     (),
     (
         Vec<(&'a Dependency, PM::PackageName, &'a Package)>,
         Vec<(&'a Dependency, PM::PackageName, &'a Package)>,
     ),
 > {
-    let graph1_edges = BTreeSet::from_iter(
-        graph1
-            .package_graph
-            .edges(pkg_name)
-            .map(|(_, pkg, dep)| (dep, pkg, graph1_pkg_table.get(&pkg).unwrap())),
-    );
-    let graph2_edges = BTreeSet::from_iter(
-        graph2
-            .package_graph
-            .edges(pkg_name)
-            .map(|(_, pkg, dep)| (dep, pkg, graph2_pkg_table.get(&pkg).unwrap())),
-    );
+    let graph1_edges =
+        BTreeSet::from_iter(graph1.package_graph.edges(pkg_name).map(|(_, pkg, dep)| {
+            (
+                dep,
+                pkg,
+                graph1_pkg_table
+                    .get(&pkg)
+                    .or_else(|| overrides.get(&pkg))
+                    .unwrap(),
+            )
+        }));
```

Comment:
Not strictly to do with this PR, but it's probably slightly neater to use `collect` rather than `BTreeSet::from_iter` here.

```suggestion
    let graph1_edges: BtreeSet<_> = graph1
        .package_graph
        .edges(pkg_name)
        .map(|(_, pkg, dep)| {
            (
                dep,
                pkg,
                graph1_pkg_table
                    .get(&pkg)
                    .or_else(|| overrides.get(&pkg))
                    .unwrap(),
            )
        })
        .collect();
```